### PR TITLE
Limit exclude search for only  NEVRA

### PR DESCRIPTION
**Pull request by j-mracek**
The issue was created by adding binary search that is enabled by default.

Closes: https://github.com/rpm-software-management/dnf5/issues/763

CI-Test: https://github.com/rpm-software-management/ci-dnf-stack/pull/1353

### CODE REVIEW

**Comment by m-blaha** [comment]
I might be worth clarifying in `man dnf5` (and in `man dnf.conf` once it's ready) that `--exclude` argument accepts `<package-name-spec>` instead of `<package-spec>`.
According to `man dnf5-specs`:
...
The `<package-spec>` argument is matched against package NEVRAs, provides and file provides.
...
`<package-name-spec>` is matched against NEVRAs only.
...

File: dnf5-plugins/changelog_plugin/changelog.cpp [hunk 1]
```diff
@@ -142,7 +142,11 @@ void ChangelogCommand::run() {
     //query
     libdnf5::rpm::PackageQuery query(ctx.base, libdnf5::sack::ExcludeFlags::APPLY_EXCLUDES, true);
     libdnf5::ResolveSpecSettings settings{
-        .ignore_case = true, .with_nevra = true, .with_provides = false, .with_filenames = false};
+        .ignore_case = true,
+        .with_nevra = true,
+        .with_provides = false,
+        .with_filenames = false,
+        .with_binaries = false};
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply dnf5-plugins/changelog_plugin/changelog.cpp:149]: There are also multiple places where also GoalJobSettings (which is a subclass of ResolveSpecSettings) is used. And on top of that, almost every Goal class method accepts GoalJobSettings argument with default value `const libdnf5::GoalJobSettings & settings = libdnf5::GoalJobSettings()`.
I guess all calls of these methods need to be checked after change of default value of `with_binaries`.

**Comment by m-blaha** [comment]
There still are places where `with_binaries=false` should have been added (like https://github.com/rpm-software-management/dnf5/blob/43eb2cd177af1c32e101678c3b229681b891568e/dnf5/context.cpp#L596).
I'm just guessing that you were searching for `ResolveSpecSettings` to find all affected places. Maybe you should search for `with_filenames` instead to cover additional occurrences.

**Comment by j-mracek** [comment]
I found additional occurrence. Thank you for suggestions.

**Comment by j-mracek** [comment]
There is a file `org.rpm.dnf.v0.rpm.Rpm.xml` that describes `with_<type>` possibilities for DBUS. Shall I update it also?

**Comment by m-blaha** [comment]
> There is a file `org.rpm.dnf.v0.rpm.Rpm.xml` that describes `with_<type>` possibilities for DBUS. Shall I update it also?

Hm, support for the `with_binaries` is not implemented yet in the dnfdaemon. So it does not make sense to add it to the documentation (which is generated from this file).

**Comment by j-mracek** [comment]
Fixed

**Review by m-blaha — APPROVED** [review]
Thans!

**Pull request merged**